src/Features/Core/Portable/Formatting/AbstractNewDocumentFormattingService.cs

```diff
@@ -41,7 +41,14 @@ public async Task<Document> FormatNewDocumentAsync(Document document, Document?
                 // other, so this shouldn't cause problems.
                 try
                 {
+                    // First we ask the provider to "format" the document. This could be formatting in terms
+                    // of adjusting block scopes to file scopes etc., but it could also be more akin to fixers
+                    // like adding access modifiers, or adding .ConfigureAwait() calls etc.
                     document = await provider.FormatNewDocumentAsync(document, hintDocument, cancellationToken).ConfigureAwait(false);
+
+                    // Now that the above has changed the document, we need the formatter to make sure its correct
+                    // before we call the next provider, otherwise they might not see things as they are meant to be.
+                    document = await Formatter.FormatAsync(document, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
❓ Format is a fairly expensive operation. Rather than running `FormatAsync`, we should run the post-processing step that the code fix engine uses (i.e. restricted to elastic trivia and nodes with FormatterAnnotation). Is there any reason we can't use that approach?

Reply:
nvm, found it